ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,280 @@
+        )
+
+
+def check_nested_elements_type(iterable, parameter_name, expected_type):
+    """Function raises ValueError exception if one of elements in collection has unexpected type"""
+    for element in iterable:
+        check_parameter_type(
+            parameter=element,
+            parameter_name=f"nested {parameter_name}",
+            expected_type=expected_type,
+        )
+
+
+def check_dictionary_keys_values_type(
+    parameter, parameter_name, expected_key_class, expected_value_class
+):
+    """Function raises ValueError exception if dictionary key or value has unexpected type"""
+    for key, value in parameter.items():
+        check_parameter_type(
+            parameter=key,
+            parameter_name=f"key in {parameter_name}",
+            expected_type=expected_key_class,
+        )
+        check_parameter_type(
+            parameter=value,
+            parameter_name=f"value in {parameter_name}",
+            expected_type=expected_value_class,
+        )
+
+
+def check_parameter_type(parameter, parameter_name, expected_type):
+    """Function extracts nested expected types and raises ValueError exception if parameter has unexpected type"""
+    # pylint: disable=W0212
+    if not isinstance(expected_type, typing._GenericAlias):  # type: ignore
+        raise_value_error_if_parameter_has_unexpected_type(
+            parameter=parameter,
+            parameter_name=parameter_name,
+            expected_type=expected_type,
+        )
+        return
+    if expected_type == typing.Any:
+        return
+    origin_class = expected_type.__dict__.get("__origin__")
+    # Checking origin class
+    raise_value_error_if_parameter_has_unexpected_type(
+        parameter=parameter,
+        parameter_name=parameter_name,
+        expected_type=origin_class,
+    )
+    # Checking nested elements
+    args = expected_type.__dict__.get("__args__")
+    if issubclass(origin_class, typing.Sequence) and args:
+        if len(args) != 1:
+            raise TypeError(
+                "length of nested expected types for Sequence should be equal to 1"
+            )
+        check_nested_elements_type(
+            iterable=parameter,
+            parameter_name=parameter_name,
+            expected_type=args,
+        )
+    elif origin_class == dict and args:
+        if len(args) != 2:
+            raise TypeError(
+                "length of nested expected types for dictionary should be equal to 2"
+            )
+        key, value = args
+        check_dictionary_keys_values_type(
+            parameter=parameter,
+            parameter_name=parameter_name,
+            expected_key_class=key,
+            expected_value_class=value,
+        )
+
+
+def check_input_param_type(checks: list):
+    """Function to apply methods on checks according to their type"""
+    for param_check in checks:
+        if isinstance(param_check, BaseInputArgumentChecker):
+            param_check.check()
```

Comment:
```suggestion
def check_input_param(*checks: BaseInputArgumentChecker):
    """Function to apply methods on checks according to their type"""
    for param_check in checks:
        if not isinstance(param_check, BaseInputArgumentChecker):
            raise TypeError(f"Wrong parameter of check_input_param: {param_check}")
        param_check.check()
```
Please, note that now the function receives `*checks` -- it means, that you can just point checks as the list of arguments, no need to enclose into `[...]`

Reply:
thanks, changed input arguments for "check_input_param" function